sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/catalog/CosmosCatalogCosmosSDKClient.scala

```diff
@@ -169,17 +169,22 @@ private[spark] case class CosmosCatalogCosmosSDKClient(cosmosAsyncClient: Cosmos
 
     private def getPartitionKeyDefinition(containerProperties: Map[String, String]): PartitionKeyDefinition = {
         val partitionKeyPath = CosmosContainerProperties.getPartitionKeyPath(containerProperties)
-
         val partitionKeyDef = new PartitionKeyDefinition
         val paths = new util.ArrayList[String]
-        paths.add(partitionKeyPath)
+        val pathList = partitionKeyPath.split(",").toList
+        if (pathList.size > 1) {
```

Comment:
Please gate the if-path on PKDefinition being >= 2 - the pkdefintion is available from CosmosConatinerProperties.getPartitionKeyVersion - the newly created PKDef should also have the same version as the original one - not hard-coded 2

Reply:
Ok, done.

Reply:
Ok - this function is called to convert a generic key/value dictionary of string representing ConatinaerProperties into a "proper" PartitionKeyDefinition. There are two or three properties working together here PKPath (which you parse form the map) - but also version and kind.

With HPK I think your code looks good - except that you "blindly" use the eixstence of KeyPath with >1 entry as trigger to set version to 2 and kind to multi-hash. What if the incoming map has version==1 or kind==hash. In the absence of those properties in the map setting them to 2 and multihash is fine - but there needs to be a violation that would result in an error if the map has incosistent values. Does that make sense?

Reply:
makes sense, changes and corresponding tests added

Reply:
LGTM now